crates/iota-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1787,6 +1805,41 @@ impl IndexerStore for PgIndexerStore {
         Ok(())
     }
 
+    async fn persist_object_versions(
+        &self,
+        object_versions: Vec<StoredObjectVersion>,
+    ) -> Result<(), IndexerError> {
+        if object_versions.is_empty() {
+            return Ok(());
+        }
+        let object_versions_count = object_versions.len();
+
+        let chunks = chunk!(object_versions, self.config.parallel_objects_chunk_size);
+        let futures = chunks
+            .into_iter()
+            .map(|c| self.spawn_blocking_task(move |this| this.persist_object_version_chunk(c)))
+            .collect::<Vec<_>>();
+
+        futures::future::join_all(futures)
```

Comment:
Nit: `try_join_all` would be more concise here.

Reply:
Looks neat, done in e3b118695cef6dd927326895996159de57987b89